### PR TITLE
Add typehints for DDPPlugin across accelerators

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

- Add/fix typehints for `DDPPlugin` across accelerators for better typechecking + IDE experience
- consistent formatting across changed files

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/ddp_cpu_slurm_accelerator.py [hunk 1]
```diff
@@ -15,16 +15,15 @@
 from typing import List, Optional
 
 import torch
-import torch.distributed as torch_distrib
 import torch.distributed as dist
```
Comment (ananthsub) [inline pytorch_lightning/accelerators/ddp_cpu_slurm_accelerator.py:18]: @SeanNaren @williamFalcon `import torch.distributed as dist` was picked due to alphabetical nature. if the rest of the codebase uses `torch_distrib` instead, I can change this

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/plugins/ddp_plugin.py [hunk 2]
```diff
@@ -20,7 +21,9 @@ def configure_ddp(self, model, device_ids):
 
     """
 
-    def configure_ddp(self, model: LightningModule, device_ids: List[int]) -> LightningModule:
+    def configure_ddp(
+        self, model: LightningModule, device_ids: List[int]
+    ) -> LightningModule:
```
Comment (ananthsub) [inline pytorch_lightning/plugins/ddp_plugin.py:26]:
```suggestion
    def configure_ddp(
        self, model: LightningModule, device_ids: List[int]
    ) -> LightningDistributedDataParallel:
```

File: pytorch_lightning/accelerators/accelerator.py [hunk 3]
```diff
@@ -35,8 +36,12 @@
 
 
 class Accelerator(object):
-
-    def __init__(self, trainer=None, cluster_environment=None, ddp_plugin=None):
+    def __init__(
+        self,
+        trainer=None,
```
Comment (Borda) [inline pytorch_lightning/accelerators/accelerator.py:41]: how about these?

File: pytorch_lightning/accelerators/accelerator.py [hunk 4]
```diff
@@ -192,7 +200,9 @@ def setup_optimizers(self, model):
         if self.trainer.testing is True:
             return
 
-        optimizers, lr_schedulers, optimizer_frequencies = self.trainer.init_optimizers(model)
+        optimizers, lr_schedulers, optimizer_frequencies = self.trainer.init_optimizers(
+            model
```
Comment (Borda) [inline pytorch_lightning/accelerators/accelerator.py:204]: what is the reason for this change? have a single arg on new line...

File: pytorch_lightning/accelerators/ddp2_accelerator.py [hunk 5]
```diff
@@ -130,7 +132,9 @@ def ddp_train(self, process_idx, mp_queue, model):
 
         """
         # show progressbar only on progress_rank 0
-        if (self.trainer.node_rank != 0 or process_idx != 0) and self.trainer.progress_bar_callback is not None:
+        if (
```
Comment (Borda) [inline pytorch_lightning/accelerators/ddp2_accelerator.py:135]: is this change needed?

File: pytorch_lightning/accelerators/ddp_spawn_accelerator.py [hunk 6]
```diff
@@ -81,15 +99,17 @@ def ddp_train(self, process_idx, mp_queue, model, is_master=False, proc_offset=0
             mp_queue: multiprocessing queue
             model:
         """
-        seed = os.environ.get("PL_GLOBAL_SEED")
+        seed = os.environ.get('PL_GLOBAL_SEED')
```
Comment (Borda) [inline pytorch_lightning/accelerators/ddp_spawn_accelerator.py:102]: why?

**Review by Borda — CHANGES_REQUESTED** [review]
it looks like Black formatting than anything else...
Black is a complete mess as one replaces " and another '
Pls make this PR as title states

File: pytorch_lightning/accelerators/accelerator.py (hunk 4 above)
Comment (williamFalcon) [reply pytorch_lightning/accelerators/accelerator.py:204]: no need to block the PR for something small like this :)
We have 80+ others to look at

File: pytorch_lightning/accelerators/ddp_cpu_slurm_accelerator.py (hunk 1 above)
Comment (williamFalcon) [reply pytorch_lightning/accelerators/ddp_cpu_slurm_accelerator.py:18]: we can do a follow up PR to make sure when using torch dist we abbreviate it as that so we don't collide with our dist

File: pytorch_lightning/accelerators/accelerator.py (hunk 4 above)
Comment (ananthsub) [reply pytorch_lightning/accelerators/accelerator.py:204]: @Borda these are all auto-formatting changes. I know it bloats the PR, but other attempts at explicitly splitting out the formatting changes from the core logic changes have also been clunky and invited questions as to whether we're black-formatting the whole codebase or not. I do strongly vouch for that, but I don't want to hold this up in the meantime. That's the main point of a standard formatter: so these PR review don't get bogged down by formatting

**Comment by Borda** [comment]
@ananthsub as it is a feature I am changing the target branch to feat 1.2, mind rebase pr and resolve conflicts? Thx your contribution and feel free to ping us back if you need something or need some help... 🐰
